src/dcap_provider.cpp

```diff
@@ -150,6 +217,43 @@ static std::string get_collateral_version()
     }
 }
 
+static std::string get_collateral_version_tdx()
+{
+    std::string collateral_version =
+        get_env_variable(ENV_AZDCAP_COLLATERAL_VER_TDX);
+
+    if (collateral_version.empty())
+    {
+        log(SGX_QL_LOG_INFO,
+            "Using tdx collateral version '%s'.",
+            tdx_collateral_version.c_str());
+
+        return tdx_collateral_version;
+    }
+    else
+    {
+        if (collateral_version.compare("v4"))
+        {
+            log(SGX_QL_LOG_ERROR,
+                "Value specified in environment variable '%s' is invalid. "
+                "Acceptable values for tdx are empty or v4",

```

Comment:
value should not be empty

Reply:
If we don't accept empty, it'd require the users to always set an environment variable before execution.